### PR TITLE
Tests

### PR DESCRIPTION
**Pull request by codydaig**
Tests too show that 0.4.0, 0.4.1, and 0.4.2 all generate

### CODE REVIEW

**Comment by lirantal** [comment]
very cool

**Comment by mleanos** [comment]
Awesome! Thank you Cody! LGTM.

File: app/index.js [hunk 1]
```diff
@@ -25,7 +25,7 @@ var versions = {
   'master': 'master',
   '0.4.0': 'v0.4.0',
   '0.4.1': 'v0.4.1',
-  '0.4.2': 'v0.4.2'
+  '0.4.2': '0.4.2'
```
Comment (ilanbiala) [inline app/index.js]: missing `v` in front?

Comment (ilanbiala) [inline app/index.js]: No, wait, that tag was published incorrectly. @lirantal the tag should in the MEAN repo should be `v0.4.2`, but instead it's `0.4.2.`

**Comment by ilanbiala** [comment]
LGTM other than the tag issue, which I'd rather just decide now before merging if we are going to fix it. I think we should for the purposes of consistency.

File: app/index.js (hunk 1 above)
Comment (codydaig) [inline app/index.js]: This was the bug fix. and I think it should be fixed and this change reverted.

Comment (lirantal) [inline app/index.js]: you're right. I updated now the tag to be v0.4.2

Comment (codydaig) [inline app/index.js]: @lirantal Did you post a link to the 0.4.2 tag anywhere? If not, let's delete it.

Comment (lirantal) [inline app/index.js]: Just on my twitter/fb page of the project, nothing too serious.
I'm ok with deleting it.

Comment (codydaig) [inline app/index.js]: Fixed in this PR as well.

**Comment by codydaig** [comment]
@ilanbiala This should be good to go now.

File: test/app.test.js [hunk 2]
```diff
@@ -64,4 +64,93 @@ describe('Main Generator', function () {
...
+    it('should have populated package.json correctly', function (done) {
+      fs.readFile('temp/package.json', function (err, data) {
+        if (err) {
+          done(err);
+        } else {
+          var packageJson = JSON.parse(data);
+          assert.equal(packageJson.name, 'meantestappname042');
+          assert.equal(packageJson.description, 'Testing the generator 042');
+          assert.equal(packageJson.author, 'Test');
+          assert.equal(packageJson['meanjs-version'], '0.4.2');
```
Comment (ilanbiala) [inline test/app.test.js]: Can we add this same assert in the other 2 tests?

File: test/app.test.js [hunk 3]
```diff
@@ -64,4 +64,93 @@ describe('Main Generator', function () {
...
+        })
+        .on('ready', function (generator) {
+          // this is called right before `generator.run()` is called
+        })
+        .on('end', function () {
+          done();
+        });
+    });
+
+    it('should generate a package.json file', function () {
```
Comment (ilanbiala) [inline test/app.test.js]: Just to make sure bower templating also works, let's check for that as well in this verification of file generation test, and then check the specific properties in the next test as well?

**Comment by ilanbiala** [comment]
@codydaig just thought of 2 small things, mostly just to make sure that we test everything that we can easily test now in case something changes.

File: test/app.test.js (hunk 2 above)
Comment (codydaig) [inline test/app.test.js]: No. This didn't get implemented until the latest release.

Comment (ilanbiala) [inline test/app.test.js]: Right, in that case can you just add a comment so that disparity is clear and not confusing in the future.

**Comment by ilanbiala** [comment]
@codydaig can you tidy up the bower checking and add the comment so we can get this merged?

**Comment by codydaig** [comment]
@ilanbiala Updated. If this looks good to you, let's merge and publish. My next task will be to help finish up one of the two other PRs for the next release.

**Comment by ilanbiala** [comment]
@codydaig better, but just for clarity when we reread these, can you add a comment saying that the meanjs-version only applies to 0.4.2 and greater? Once you commit that, you can merge. Ping me, and I'll release.

**Comment by codydaig** [comment]
@ilanbiala Ready to publish!

**Pull request merged**